File: CITS3002 - Networks/Project - Transport Station - Client and Server/C++Project/RouteStop.cpp

#include "globals.h"

/**
 * Create a route stop from parts
 * @param nameIn
 * @param portIn
 */
RouteStop::RouteStop(String nameIn, int portIn) {
    name = nameIn;
    port = portIn;
}

/**
 * Create a route stop from a String rep
 * @param nameIn
 * @param portIn
 */
RouteStop::RouteStop(String strRep) {
    vector<String> parts = split(strRep, sep);
    name = parts.at(0);
    port = stoi(parts.at(1));
}

/**
 * Get the String rep of a route stop
 * @return 
 */
String RouteStop::toString() {
    return (name + sep + to_string(port));
}

/**
 * Return true if this RS is as same the input
 * @param comp
 * @return 
 */
bool RouteStop::equals(RouteStop comp) {
    bool samePort = port == comp.getPort();
    bool sameName = strEquals(name, comp.getName());
    return sameName && samePort;
}

/**
 * Update the port's stop
 *
 * @param newPort
 */
void RouteStop::setPort(int newPort) {
    port = newPort;
}

/**
 * Get stop name
 *
 * @return
 */
String RouteStop::getName() {
    return name;
}

/**
 * Get stop port
 *
 * @return
 */
int RouteStop::getPort() {
    return port;
}

// ----
// ------------------------------------
// ----

/**
 * Default value
 */
RouteStop::RouteStop() {
    name = "Default";
    port = -1;
}

/**
 * Copy constructor - enables use in vectors
 * @param orig
 */
RouteStop::RouteStop(const RouteStop& orig) {
    name = orig.name;
    port = orig.port;
}

/**
 * Destructor
 */
RouteStop::~RouteStop() {
}

File: CITS3002 - Networks/Project - Transport Station - Client and Server/JavaProject/src/statgram/log/RouteStop.java
package statgram.log;

import java.util.Objects;

/**
 * Represents a stop in a route
 *
 * @author David
 */
public class RouteStop {

    // Separator
    private final String sep = "~~";

    // The station's name
    private final String name;

    // The station's port
    private int port;

    /**
     * Create a route stop from values
     *
     * @param name
     * @param port
     */
    public RouteStop(String name, int port) {
        this.name = name;
        this.port = port;
    }

    /**
     * Create a route stop from a string representation
     *
     * @param strRep
     */
    public RouteStop(String strRep) {
        String[] parts = strRep.split(sep);
        name = parts[0];
        port = Integer.parseInt(parts[1]);
    }

    /**
     * Get string representation
     *
     * @return
     */
    @Override
    public String toString() {
        return (name + sep + port);
    }

    /**
     * Replacement for equals function
     *
     * @param obj
     * @return
     */
    @Override
    public boolean equals(Object obj) {
        RouteStop comp = null;
        if (obj instanceof RouteStop) {
            comp = (RouteStop) obj;
        } else {
            throw new IllegalArgumentException("RouteStop equals");
        }
        boolean samePort = this.port == comp.getPort();
        boolean sameName = this.name.equals(comp.getName());
        return sameName && samePort;
    }

    /**
     * Replacement for hash code function
     *
     * @return
     */
    @Override
    public int hashCode() {
        int hash = 7;
        hash = 53 * hash + Objects.hashCode(this.sep);
        hash = 53 * hash + Objects.hashCode(this.name);
        hash = 53 * hash + this.port;
        return hash;
    }

    /**
     * Update the port
     *
     * @param newPort
     */
    public void setPort(int newPort) {
        port = newPort;
    }

    /**
     * Get station name
     *
     * @return
     */
    public String getName() {
        return name;
    }

    /**
     * Get station port
     *
     * @return
     */
    public int getPort() {
        return port;
    }

}
